### PR TITLE
Fix for issue #419

### PR DESCRIPTION
**Pull request by OsirisTerje**
*** DO NOT MERGE **

Issue #419 is partially fixed.

Updated to NUnit 3.9.0
Updated/consolidated other packages.
Fixed attachment test case.

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -12,7 +12,7 @@ var configuration = Argument("configuration", "Release");
 //////////////////////////////////////////////////////////////////////
 
 var version = "3.10.0";
-var modifier = "";
+var modifier = ".2";
```
Comment (OsirisTerje) [inline build.cake]: I know........

Comment (CharliePoole) [reply build.cake]: You seem to be using modifier in a different way than intended. Is this what you have done all along? Normally, I would have said version = "3.10.0.2" and modifier ="" (or "beta" or "rc-2" etc.).

File: src/NUnitTestAdapter/TestConverter.cs [hunk 2]
```diff
@@ -116,8 +116,15 @@ public IList<VSTestResult> GetVSTestResults(XmlNode resultNode)
...
             }
 
-            return results;
+            return new TestResultSet {TestCaseResult = testcaseResult, TestResults = results};
+        }
+
+        public struct TestResultSet
+        {
+            public IList<VSTestResult> TestResults { get; set; }
+            public TestResult TestCaseResult { get; set; }
```
Comment (CharliePoole) [inline src/NUnitTestAdapter/TestConverter.cs]: This all gets a bit confusing without a prefix to distinguish NUnit from VS results.

File: build.cake (hunk 1 above)
Comment (OsirisTerje) [reply build.cake]: I use the modifier to get an increasing number, then I use the configuration Debug, which causes the script to add the suffix '-dbg'   I only use this during development, and clear this before commit, which I just forgot here.    If you set modifier = beta,  you get 3.10.0beta-dbg, note the missing dot between 0 and beta.

File: src/NUnitTestAdapter/TestConverter.cs (hunk 2 above)
Comment (OsirisTerje) [reply src/NUnitTestAdapter/TestConverter.cs]: Good point, I'll fix.

**Comment by OsirisTerje** [comment]
If you look in the method TestFinished, you'll see I get the result from the testcase, and use that for the  overall test result outcome, then I just add the assertion results as result outputs, but they don't change the outcome.  
I work by the assumption you stated earlier, that NUnit will always return the outcome for the testcase, instead of checking all the assertions.
(If this assumption don't hold, we miss a testcase for that)

**Comment by CharliePoole** [comment]
I think relying on NUnit's outcome for each test is all you can do. If NUnit itself is wrong, then we would have to fix it. WRT the individual assertion results, I could not see where you were getting them in your changes - is this in the part you didn't change?

File: build.cake (hunk 1 above)
Comment (CharliePoole) [reply build.cake]: I see, of course there's no problem with what you do in development.

With the modifiers, we actually used "-beta" or "-rc-2" when we were using them.

**Comment by OsirisTerje** [comment]
Yes, that code is unchanged, each assertion is added as a result to the result collection, plus a recordresult is sent to VS.

**Comment by OsirisTerje** [comment]
No further comments, so I'll merge it.

**Pull request merged**